### PR TITLE
feat(demo-app): init

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

init demo app project

## Linear Issue Reference

LOG-2747

## Testing

`pnpm start` works

### CODE REVIEW

File: packages/demo-app/package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "prettier": "^2.3.2",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "stylelint": "^14.8.2",
+    "typescript": "^4.7.2"
+  },
+  "eslintConfig": {
+    "extends": "@silverhand/react",
+    "rules": {
+      "complexity": "off"
```
Comment (demonzoo) [inline packages/demo-app/package.json:41]: nit: How about putting this rule in `eslint-config-react`?

Comment (gao-sun) [reply packages/demo-app/package.json:41]: still thinking about this. or we can allow more complexity in tsx files

Comment (demonzoo) [reply packages/demo-app/package.json:41]: Yep, both are fine

File: packages/demo-app/src/index.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="utf-8" />
+    <title>Logto Console</title>
```
Comment (demonzoo) [inline packages/demo-app/src/index.html:6]: Logto Demo App

**Review by demonzoo — APPROVED** [review]

File: packages/demo-app/package.json (hunk 1 above)
Comment (gao-sun) [reply packages/demo-app/package.json:41]: let's do it later

File: packages/shared/declaration/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/* eslint-disable import/no-unassigned-import */
+import './react-app';
+import './dom';
+/* eslint-enable import/no-unassigned-import */
```
Comment (demonzoo) [inline packages/shared/declaration/index.ts:4]: We can actually import all the global scss files from this shared package, right? Like `normalized.scss`

Comment (gao-sun) [reply packages/shared/declaration/index.ts:4]: discussed offline, do it later

**Pull request merged**